File: API/API/services/ITaskService.vb
Public Interface ITaskService
    Function CreateTask(req As CreateTaskRequest, creatorId As Guid) As String

    Function UpdateTask(req As UpdateTaskRequest, currentUserId As Guid) As String

    Function DeleteTask(taskId As Integer, currentUserId As Guid) As String

    Function AssignTask(req As AssignTaskRequest, currentUserId As Guid) As String
    Function GetMyCreatedTasks(userId As Guid) As List(Of TaskDetailResponse)

    Function GetMyAssignedTasks(userId As Guid) As List(Of TaskResponse)

    Function UpdatePermission(req As AssignTaskRequest, currentUserId As Guid) As String

    Function RemoveAssignment(req As RemoveAssignmentRequest, currentUserId As Guid) As String
    Sub ChangeStatus(req As ChangeTaskStatusRequest, currentUserId As Guid)
End Interface

File: API_v2/Services/Interfaces/ITaskService.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using API_v2.Models.DTOs;

namespace API_v2.Services.Interfaces
{
    public interface ITaskService
    {
        Task<string> CreateTaskAsync(CreateTaskRequest req, Guid creatorId, Guid projectId);
        Task<string> UpdateTaskAsync(int taskId, UpdateTaskRequest req, Guid currentUserId);
        Task<string> DeleteTaskAsync(int taskId, Guid currentUserId);
        Task<string> AssignTaskAsync(AssignTaskRequest req, Guid currentUserId);
        Task<PagedResponse<TaskDetailResponse>> GetProjectTasksAsync(Guid projectId, Guid userId, int? columnId, int page, int pageSize, string search = null, API_v2.Models.Enums.TaskPriority? priority = null, Guid? assigneeId = null);
        Task<TaskStatsResponse> GetTaskStatsAsync(Guid projectId, Guid userId);
        Task<string> RemoveAssignmentAsync(int taskId, Guid userId, Guid currentUserId);
        Task ChangeColumnAsync(ChangeTaskColumnRequest req, Guid currentUserId);
    }
}
